npep/npep-126-egress-traffic-control.md

```diff
@@ -0,0 +1,100 @@
+  workloads (non-hostNetworked pods) towards hostNetworked pods in the
+  cluster is undefined. See https://github.com/kubernetes-sigs/network-policy-api/issues/73.
+  - ANP currently supports only east-west traffic and this traffic flow cuts from
+  overlay to underlay which makes this part of the egress (northbound) use case.
+  - Let's provide a defined behaviour in ANP to explicitly achieve the use case.
+  - NOTE: We can keep this goal experimental in case implementors cannot distinguish between
+    a node and host-networked pod
+
+## Non-Goals
+
+* Implementing southbound traffic use cases is outside the scope of this NPEP
+* Implementing specific api-server user stories where the definition of an api-server
+matches neither a nodeSelector not a podSelector or any expressive external destinaton
+is outside the scope of thie NPEP and perhaps warrants a new NPEP.
+* Implementing traffic towards nodes exluding traffic towards host-networked pods on that
+node is outside the scope of this NPEP and is a non-goal
+
+## Introduction
+
+### User Stories for egress traffic control towards external destinations
+
+1. **As a** cluster administrator **I want** to restrict traffic from
+specific cluster workloads to all or specific destinations outside the
+cluster **so that** I can enforce security for northbound traffic.
+Example: Pods in namespaceA and namespaceB should not be able to talk
+to the internet but they should be able to access company's intranet.
+
+2. **As a** cluster administrator **I want** to explicitly allow traffic
+from cluster workloads to handpicked destinations outside the cluster and
+deny the rest **so that** I can guarantee cluster wide egress traffic
+flow irrespective of whatever network policies are defined by the
+developer for specific namespaces. Example: As an owner of namespaceA I
+define policies that deny all northbound egress traffic for that
+namespace. However the cluster-admin can decide all namespaces in the
+cluster must be able to talk to the EXTERNAL_DNS_SERVER_IP on port 53.
+
+3. **As a** cluster administrator **I want** to explicitly delegate policy
+decisions for traffic from specific cluster workloads to all or specific
+destinations outside the cluster to the namespace owners or application
+developers **so that** the existing network policies that match these same set
+of destinations (a.k.a `ipBlock` in NetworkPolicyV1 world) can take effect
+Example: Pods in namespaceA should not be able to talk to 8.8.8.8 according
+to the administrator but he is not sure and thus wants to `Pass` this decision
+to the network policies defined by the namespaceA owner instead.
+
+### User Stories for egress traffic control towards host-networked pods
+
+1. **As a** cluster administrator **I want** to easily block access from
+cluster workloads to k8s-apiservers running on cluster nodes as host-networked pods
+but continue to allow traffic to the rest of the node.
```

Comment:
unclear what "the rest of the node" refers to. (Other host-networked pods? Pod-network pods?)

Reply:
no idea lol, i think that last part doesn't make sense...removing it